kratos/solving_strategies/strategies/explicit_solving_strategy_runge_kutta_4.h

```diff
@@ -0,0 +1,414 @@
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ExplicitSolvingStrategyRungeKutta4";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream &rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream &rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    void SolveWithLumpedMassMatrix() override
+    {
+        // Get the required data from the explicit builder and solver
+        const auto p_explicit_bs = BaseType::pGetExplicitBuilder();
+        auto& r_dof_set = p_explicit_bs->GetDofSet();
+        const unsigned int dof_size = p_explicit_bs->GetEquationSystemSize();
+        const auto& r_lumped_mass_vector = p_explicit_bs->GetLumpedMassMatrixVector();
+
+        // Set the auxiliary RK vectors
+        LocalSystemVectorType u_n(dof_size); // TODO: THIS IS INEFICCIENT. CREATE A UNORDERED_SET WITH THE IDOF AND VALUE AS ENTRIES. THIS HAS TO BE OPTIONAL
+        LocalSystemVectorType rk_k1(dof_size);
+        LocalSystemVectorType rk_k2(dof_size);
+        LocalSystemVectorType rk_k3(dof_size);
+        LocalSystemVectorType rk_k4(dof_size);
+
+        // Perform the RK 4 update
+        const double dt = BaseType::GetDeltaTime();
+        KRATOS_ERROR_IF(dt < 1.0e-12) << "ProcessInfo DELTA_TIME is close to zero." << std::endl;
+
+        // Set the previous step solution in the current buffer position
+        // Note that we set the 0 position of the buffer to be equal to the values in step n (not n+1)
+        // Additionally, we save in an auxiliary vector the value of the fixed DOFs, which is also taken from the previous time step
+        IndexPartition<int>(r_dof_set.size()).for_each(
+            [&](int i_dof){
+                auto it_dof = r_dof_set.begin() + i_dof;
+                double& r_u_0 = it_dof->GetSolutionStepValue(0);
+                const double& r_u_1 = it_dof->GetSolutionStepValue(1);
+                if (it_dof->IsFixed()) {
+                    u_n(i_dof) = r_u_1;
+                }
+                r_u_0 = r_u_1;
+            }
+        );
+
+        // Calculate the RK4 intermediate sub steps
+        PerformRungeKuttaIntermediateSubStep(1, mA21, u_n, rk_k1);
+        PerformRungeKuttaIntermediateSubStep(2, mA32, u_n, rk_k2);
+        PerformRungeKuttaIntermediateSubStep(3, mA43, u_n, rk_k3);
+        PerformRungeKuttaLastSubStep(rk_k4);
+
+        // Do the final solution update
+        IndexPartition<int>(r_dof_set.size()).for_each(
```

Comment:
same

Reply:
Done.

Reply:
Is required to access the mass contribution.

Reply:
As well as de BC value if the DOF is fixed.